### ISSUE TITLE
[0.23.0-rc1] Cannot resolve react-native

### ISSUE THREAD

**Issue by IanVS** [facebook/react-native#6640]
On mac, npm v.3.8.2, node 5.3.0:

On a fresh install:

```
npm i --save react react-native@next
```

The install seems to complete successfully. 

```
npm ls --depth=0
rn@1.0.0 /Users/ianvs/code/experiments/rn
├── react@0.14.7
└── react-native@0.23.0-rc1
```

But:

```
npm ls react
rn@1.0.0 /Users/ianvs/code/experiments/rn
└── react@0.14.7

npm ls react-native
rn@1.0.0 /Users/ianvs/code/experiments/rn
└── (empty)

npm ERR! code 1
```

This is causing problems in some third party libraries I use which import RN with `var React = require('react-native`);`.  

![image](https://cloud.githubusercontent.com/assets/4616705/14031744/b33cc168-f1e4-11e5-955f-876ef33ba5ed.png)

I had no trouble with `0.22.2`.

**Comment by xiaobuu** [comment]
Same here. After upgrading to 0.23-rc1, on android, a third party module throws unable to resolve module Subscribable. Everything works on iOS, and works on 0.22.2 and before.

**Comment by rogchap** [comment]
Do you have Hot Reloading Enabled? I had this issue and switching off Hot Reloading "solved" it.

**Comment by brentvatne** [comment]
I had the same issue with HMR enabled @rogchap

**Comment by IanVS** [comment]
Surprisingly yes, disabling hot reloading did prevent the redboxes (both Android and iOS).  Is the `npm ls` behavior a red herring?

**Comment by cpojer** [comment]
cc @martinbigio

**Comment by martinbigio** [comment]
Hmm I cannot repro neither on master nor @next. Doesn't make sense that this repro only with HMR enabled. @brentvatne, @IanVS could you guys try reseting the packagers cache?

@satya164, have you seen this issue?

**Comment by IanVS** [comment]
I used `npm start -- --clear-cache` with no change in behavior.

**Comment by martinbigio** [comment]
@IanVS you should run `npm start -- --reset-cache` (`reset-cache` instead of `clear-cache`).

**Comment by IanVS** [comment]
No luck.  I'll try to create a minimal reproduction.

**Comment by IanVS** [comment]
I can reproduce on a clean install with:

```
react-native init repro
cd repro
npm i --save react-native-vector-icons react-native@next
```

Add the following to the top of `index.ios.js`:

```js
import Icon from 'react-native-vector-icons';
```

Starting it up should work fine, getting the default "Welcome to React Native~" screen.

```
react-native run-ios
```

But, if you enable hot reloading, you should see a redbox similar to the one I posted above.

**Comment by martinbigio** [comment]
Thanks @IanVS, I can repro following your steps.

The require on the HMR path happens here: https://github.com/facebook/react-native/blob/3a1634ec7b0dfa2b5d8890118565006d845e2039/babel-preset/configs/hmr.js#L28.

Not sure why this broke but changing React for react-native should be equivalent and it works. Will submit PR and publish a new version of the preset soon.

**Comment by satya164** [comment]
Only happens when you install some other pacakge? Weird.

**Comment by martinbigio** [comment]
Should work now that we have released babel-preset-react-native@1.5.5. You'll have to upgrade from 1.5.4 to 1.5.5
